### PR TITLE
Mostly php cleanup

### PR DESCRIPTION
**Pull request by GregOriol**
This is mostly php cleanup:
- cleanup of syntax to match current standards
- adding phpdoc to methods and objects, which gives nice IDE completion/hints and eases using the code from other projects (still some real documentation to do though)
- adding explicit exception types, which allows to catch errors
- also fixes php fatal error on parsing an exception response from killbill

### CODE REVIEW

**Comment by MaxGfeller** [comment]
Thanks for the mention @pierre.

@GregOriol What standards are you referring to? The current active community standard is [`PSR-2`](https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-2-coding-style-guide.md) which i think would be great to adapt.

It would be cool to be able to run a linter using a composer script, which maybe also should be run in `composer test`.

I like the custom Exception class but i think it would be even better to have multiple ones, e.g. `UnauthorizedException`, `ConnectionTimeoutException` etc. They can be more fine-tuned to be able to react better to them.

**Comment by GregOriol** [comment]
@MaxGfeller Yes indeed PSR-2, and also a little bit of Symfony

I added phpcs with those rules (composer phpcs)

Indeed more exceptions would be nice, I'm still figuring out which ones might be useful, best bet I guess would be to implement more when needed

**Comment by pierre** [comment]
> Thanks for the mention @pierre.

Since it's quite a large diff, I just want to make sure it doesn't affect your integration, and make sure it goes into the right direction after the recent work you did.

Let me know when you feel it's good to be merged. Thanks in advance for the review!

File: composer.json [hunk 1]
```diff
@@ -3,23 +3,27 @@
...
+    }
   },
   "scripts": {
-      "test": "ENV=local phpunit",
-      "test-integration": "phpunit",
-      "record-mocks": "composer run test-integration && RECORD_REQUESTS=1 phpunit"
+    "test": "ENV=local phpunit",
+    "test-integration": "phpunit",
+    "record-mocks": "composer run test-integration && RECORD_REQUESTS=1 phpunit",
+    "phpcs": "./vendor/squizlabs/php_codesniffer/scripts/phpcs --standard=vendor/escapestudios/symfony2-coding-standard/Symfony2 ./src/ ./test/ --extensions=php --ignore=src/Type -d memory_limit=1024M",
```
Comment (MaxGfeller) [inline composer.json]: You can use here just `phpcs` instead of the whole path. `vendor/bin/` is being prepended to the `$PATH` by Composer.

**Comment by GregOriol** [comment]
@MaxGfeller I also just added the re-generated types to match the cleanup (after https://github.com/killbill/killbill-java-parser/pull/12)

**Comment by MaxGfeller** [comment]
> Indeed more exceptions would be nice, I'm still figuring out which ones might be useful, best bet I guess would be to implement more when needed

I think it would be good to have "standard API client" ones already, like the Unauthorized or Unreachable one. And of course we could add more over time. Maybe those will help: https://github.com/killbill/killbill-client-ruby/blob/67cbd3c801b18890c9d42d19ca0efe45cd8bcce7/spec/killbill_client/errors_spec.rb

IMO it'd be cool if the coding style would be checked by travis. We could enhance the `.travis.yml` like this:

```
script: composer phpcs && composer test
```

What do you think?

**Comment by GregOriol** [comment]
@MaxGfeller Seems a great idea, but I won't have time to work on the exceptions now

The phpcs report currently contains some errors due to missing class doc blocks and some other things. Maybe travis integration can wait a little bit? or produce non-blocking reports at least

**Comment by MaxGfeller** [comment]
@GregOriol IMO those errors should be resolved as quickly as possible. 

I can work on the Exceptions and on the phpcs errors, but i probably can't commit to this pull request. So it'd be fine for me to have this merged and i'll start my work in an additional pull request.

What do you think @pierre?

@GregOriol The tests are failing now. Do the integration tests work? Maybe you'd have to re-record the mocks, see [this issue](https://github.com/killbill/killbill-client-php/pull/15).

**Comment by GregOriol** [comment]
Tests should be OK now, sorry for that

Imo some feedback from phpcs are a nice to have, compared to some features still missing from the client; so for my part, I'd prefer focusing on the features :-)

Anyway, if the PR is merged, other changes and enhancements can be made later; you could probably create an issue to log the ideas, and we'll catch up on them later?

**Comment by MaxGfeller** [comment]
I'll start working on these things as soon as your pull request is merged :)

So for me it's good to go, @pierre

**Comment by pierre** [comment]
Thanks guys!

**Pull request merged**